File: ios/Classes/AssetsAudioPlayerPlugin.m
#import "AssetsAudioPlayerPlugin.h"
#import <assets_audio_player/SwiftAssetsAudioPlayerPlugin.swift-Swift.h>

@implementation AssetsAudioPlayerPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftAssetsAudioPlayerPlugin registerWithRegistrar:registrar];
}
@end

File: macos/Classes/AssetsAudioPlayerPlugin.swift
import Cocoa
import FlutterMacOS

public class AssetsAudioPlayerPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "assets_audio_player", binaryMessenger: registrar.messenger)
    let instance = AssetsAudioPlayerPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
